lib/http/HttpClient_WinInet.cpp

```diff
@@ -316,19 +318,13 @@ class WinInetRequestWrapper
 
     void onRequestComplete(DWORD dwError)
     {
-        std::unique_ptr<SimpleHttpResponse> response(new SimpleHttpResponse(m_id));
-
-        std::vector<uint8_t> & m_bodyBuffer = response->m_body;
-        DWORD m_bufferUsed = 0;
-
         if (dwError == ERROR_SUCCESS) {
             // If looking good so far, try to fetch the response body first.
             // It might potentially be another async operation which will
             // trigger INTERNET_STATUS_REQUEST_COMPLETE again.
 
             m_bodyBuffer.insert(m_bodyBuffer.end(), m_buffer, m_buffer + m_bufferUsed);
             do {
```

Comment:
Do not need to enter this loop (more accurately: prevent calling InternetReadFile again) if the previous call to IRF returned ERROR_IO_PENDING followed by a zero-byte buffer on completion -- and this function is called on completion of the previous IRF.

Reply:
good call, moved into the if block below

Reply:
Added the if statement, is this what you mean?

Reply:
Changed to !m_responsePending || m_bufferUsed != 0 like suggested